docs/components/popup.md

```diff
@@ -1,73 +1,70 @@
 
 ```js
-import ReactMap, {Popup} from 'react-map-gl';
-
-export default MyMap = () => {
- return (
-  <ReactMap latitude={-122.41} longitude={37.78} zoom={8}>
-   <Popup latitude={-122.41} longitude={37.78}
-     closeButton={true} closeOnClick={false}
-     anchor="top" >
-    <div>You are here</div>
-   </Popup>
-  </ReactMap>
- );
+import {Component} from 'react';
+import ReactMapGL, {Popup} from 'react-map-gl';
+
+class Map extends Component {
+  render() {
+    return (
+      <ReactMapGL latitude={-122.41} longitude={37.78} zoom={8}>
+        <Popup latitude={-122.41} longitude={37.78} closeButton={true} closeOnClick={false} anchor="top">
+          <div>You are here</div>
+        </Popup>
+      </ReactMapGL>
+    );
+  }
 }
 ```
 
-## Styling
-
-Like its Mapbox counterpart, this control relies on the mapbox-gl stylesheet to work properly.
-
-You may add the stylesheet to your page:
-```
-<!-- index.html -->
-<link href='https://api.tiles.mapbox.com/mapbox-gl-js/v0.37.0/mapbox-gl.css' rel='stylesheet' />
-```
-
-Or embed it in your app by using [browserify-css](https://www.npmjs.com/package/browserify-css) with Browserify or [css-loader](https://webpack.github.io/docs/stylesheets.html) with Webpack:
-```
-/// app.js
-import 'mapbox-gl/dist/mapbox-gl.css';
-```
-
-
 ## Properties
 
-##### `latitude` (Number)
+##### `latitude` {Number} (required)
 Latitude of the marker.
 
-##### `longitude` (Number)
+##### `longitude` {Number} (required)
 Longitude of the marker.
 
-##### `offsetLeft` (Number, optional)
+##### `offsetLeft` {Number} - default: `0`
 Offset of the marker from the left in pixels, negative number indicates left.
 
-##### `offsetTop` (Number, optional)
+##### `offsetTop` {Number} - default: `0`
 Offset of the marker from the top in pixels, negative number indicates up.
 
-##### `closeButton` (Boolean, optional)
+##### `closeButton` {Boolean} - default: `true`
 If `true`, a close button will appear in the top right corner of the popup.
-Default `true`.
 
-##### `closeOnClick` (Boolean, optional)
+##### `closeOnClick` {Boolean} - default: `true`
 If `true`, the popup will closed when the map is clicked.
-Default `true`.
 
-##### `tipSize` (Number, optional)
+##### `tipSize` {Number} - default: `10`
 Size of the tip pointing to the coordinate.
-Default `10`.
 
-##### `anchor` (String, optional)
+##### `anchor` {String} - default: `bottom`
 A string indicating the popup's position relative to the coordinate.
 Options are `top`, `bottom`, `left`, `right`, `top-left`, `top-right`, `bottom-left`, and `bottom-right`.
 
-##### `dynamicPosition` (Boolean, optional)
+##### `dynamicPosition` {Boolean} - default: `true`
 If `true`, the anchor will be dynamically adjusted to ensure the popup falls within the map container.
-Default `true`.
 
-##### `onClose` (Function, optional)
+##### `onClose` {Function}
 Callback when the user closes the popup.
+
+## Styling
```

Comment:
Happy to switch to that but I'll let it formulates subsequently, outside of this PR.